src/test/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapperTest.java

```diff
@@ -606,10 +606,39 @@ public void setFinalTrackingIdTest() throws IOException, CsafException {
         advisory.setFinalTrackingIdAndUrl("https://example.com", "example", "5", 158L);
         long year = ZonedDateTime.now().getYear();
         assertEquals("example-" + year + "-00158", advisory.getDocumentTrackingId());
-        assertEquals("https://example.com/WHITE/" + year + "/example-" + year + "-00158.json", advisory.at("/csaf/document/references/0/url").asText());
+        assertEquals("https://example.com/white/" + year + "/example-" + year + "-00158.json", advisory.at("/csaf/document/references/0/url").asText());
         assertEquals("URL generated by system", advisory.at("/csaf/document/references/0/summary").asText());
         assertEquals("self", advisory.at("/csaf/document/references/0/category").asText());
         assertEquals("tempExamle-TEMP-0000123", advisory.getTempTrackingIdInFromMeta());
     }
 
+    @Test
+    @SuppressFBWarnings(value = "CE_CLASS_ENVY", justification = "Only for Test")
+    public void setFinalTrackingIdTest_label_not_null() throws IOException, CsafException {
+
+        final String TLP_LABEL = "AMBER";
+
+        var csafJsonWithReleaseDate = """
+                { "document": {
+                     "distribution": {
+                         "tlp": {
+                           "label": "%s"
+                         }
+                       },
+                      "publisher": {
+                         "name": "Red flag company"
+                      }
+                   }
+                }""".formatted(TLP_LABEL);
+
+        AdvisoryWrapper advisory = AdvisoryWrapper.createNewFromCsaf(csafToRequest(csafJsonWithReleaseDate), "Mustermann", Semantic.name());
+        advisory.setTemporaryTrackingId("tempExamle", "7", 123L);
+        advisory.setFinalTrackingIdAndUrl("https://example.com", "example", "5", 158L);
+        long year = ZonedDateTime.now().getYear();
+        assertEquals("example-" + year + "-00158", advisory.getDocumentTrackingId());
+        assertEquals("https://example.com/" + TLP_LABEL.toLowerCase() + year + "/example-" + year + "-00158.json", advisory.at("/csaf/document/references/0/url").asText());
```

Comment:
I think a “/” is missing
```suggestion
        assertEquals("https://example.com/" + TLP_LABEL.toLowerCase() + "/" + year + "/example-" + year + "-00158.json", advisory.at("/csaf/document/references/0/url").asText());
```